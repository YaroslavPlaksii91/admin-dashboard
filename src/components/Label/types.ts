import { LABEL_COLORS, LABEL_TYPES } from './constants';

export type LabelProps = {
  type: (typeof LABEL_TYPES)[keyof typeof LABEL_TYPES];
  color: (typeof LABEL_COLORS)[keyof typeof LABEL_COLORS];
  text: string;
};
