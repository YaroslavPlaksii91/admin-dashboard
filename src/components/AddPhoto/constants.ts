export const IMAGE_SIZE = Object.freeze({
  SMALL: 44,
  BIG: 60,
});
