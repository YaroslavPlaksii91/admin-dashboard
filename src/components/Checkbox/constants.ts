export const CHECKBOX_TEST_ID = 'checkbox';
