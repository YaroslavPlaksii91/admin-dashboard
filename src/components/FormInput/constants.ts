export const TOGGLE_BUTTON_TEST_ID = 'toggleButton';
