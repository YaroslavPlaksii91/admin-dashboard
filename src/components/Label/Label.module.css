.label {
  padding: 5px 12px;
  font-size: 11px;
  font-weight: 700;
  line-height: normal;
  letter-spacing: 0.5px;
  text-transform: uppercase;
  color: var(--main-white-color);
}

.round {
  border-radius: 100px;
}

.square {
  border-radius: 8px;
}

.default {
  color: var(--secondary-text-color);
  background-color: var(--input-border-color);
}

.new,
.normal {
  background-color: var(--green-color);
}

.urgent,
.low {
  background-color: var(--yellow-color);
}

.high {
  background-color: var(--red-color);
}
