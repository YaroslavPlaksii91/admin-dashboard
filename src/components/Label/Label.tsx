import { FC } from 'react';
import classNames from 'classnames';

import { LabelProps } from './types';

import styles from './Label.module.css';

export const Label: FC<LabelProps> = ({ type, color, text }) => {
  const labelClassNames = classNames(styles.label, {
    [styles[type]]: type,
    [styles[color]]: color,
  });

  return <span className={labelClassNames}>{text}</span>;
};
