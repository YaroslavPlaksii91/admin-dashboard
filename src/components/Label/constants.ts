export const LABEL_TYPES = Object.freeze({
  ROUND: 'round',
  SQUARE: 'square',
});

export const LABEL_COLORS = Object.freeze({
  DEFAULT: 'default',
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error',
});
