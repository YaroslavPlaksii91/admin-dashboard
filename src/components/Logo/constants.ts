export const LOGO_TYPES = Object.freeze({
  VERTICAL: 'vertical',
  HORIZONTAL: 'horizontal',
});
