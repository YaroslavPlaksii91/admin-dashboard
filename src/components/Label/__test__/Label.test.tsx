import { cleanup, render, screen } from '@testing-library/react';

import { Label } from '../Label';

import { LABEL_COLORS, LABEL_TYPES } from '../constants';

describe('Label component', () => {
  afterEach(cleanup);

  it('should be rendered', () => {
    const { asFragment } = render(
      <Label
        color={LABEL_COLORS.DEFAULT}
        type={LABEL_TYPES.ROUND}
        text="Label"
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with the correct text', () => {
    render(
      <Label
        color={LABEL_COLORS.DEFAULT}
        type={LABEL_TYPES.ROUND}
        text="Label"
      />,
    );

    const labelElement = screen.getByText('Label');

    expect(labelElement).toBeInTheDocument();
  });
});
