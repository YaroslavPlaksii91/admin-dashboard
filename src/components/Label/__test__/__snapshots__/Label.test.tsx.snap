// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should be rendered 1`] = `
<DocumentFragment>
  <span
    class="undefined"
  >
    Label
  </span>
</DocumentFragment>
`;
